src/modules/swap/services/safeBundleFlow/index.ts

```diff
@@ -63,6 +65,27 @@ export async function safeBundleFlow(
       { to: presignTx.to!, data: presignTx.data!, value: '0', operation: 0 },
     ]
 
+    const shouldZeroApprove = await shouldZeroApproveFn({
```

Comment:
I see the code duplicates for swap and limit-orders.
@alfetopito maybe we can consolidate it in the future

Reply:
100%, actually i was wondering the same when I saw it in limit orders. 

I guess your comment is more general, and maybe you mean to refactro the flows themselve too